src/rua_environment.rs

```diff
@@ -49,51 +56,56 @@ pub fn prepare_environment(config: &CliArgs) {
 		env!("CARGO_PKG_NAME"),
 		env!("CARGO_PKG_VERSION")
 	);
-	assert!(
-		env::var_os("PKGDEST").is_none(),
-		"Cannot work with PKGDEST environment being set. Please run RUA without it"
-	);
-	env::set_var("PKGDEST", "/dev/null"); // make sure it's overridden later
-	assert!(
-		env::var_os("SRCDEST").is_none(),
-		"Cannot work with SRCDEST environment being set. Please run RUA without it"
-	);
-	env::set_var("SRCDEST", "/dev/null"); // make sure it's overridden later
-	assert!(
-		env::var_os("SRCPKGDEST").is_none(),
-		"Cannot work with SRCPKGDEST environment being set. Please run RUA without it"
-	);
-	env::set_var("SRCPKGDEST", "/dev/null"); // make sure it's overridden later
-	assert!(
-		env::var_os("LOGDEST").is_none(),
-		"Cannot work with LOGDEST environment being set. Please run RUA without it"
-	);
-	env::set_var("LOGDEST", "/dev/null"); // make sure it's overridden later
-	assert!(
-		env::var_os("BUILDDIR").is_none(),
-		"Cannot work with BUILDDIR environment being set. Please run RUA without it"
-	);
-	env::set_var("BUILDDIR", "/dev/null"); // make sure it's overridden later
-	if let Some(extension) = std::env::var_os("PKGEXT") {
-		assert!(
-			extension == ".pkg.tar"
-				|| extension == ".pkg.tar.xz"
-				|| extension == ".pkg.tar.lzma"
-				|| extension == ".pkg.tar.gz"
-				|| extension == ".pkg.tar.gzip"
-				|| extension == ".pkg.tar.zst"
-				|| extension == ".pkg.tar.zstd",
-			"PKGEXT environment is set to an incompatible value. \
-			 Only .pkg.tar or .pkg.tar.xz or .pkg.tar.gz or .pkg.tar.zst archives are supported for now.\
-			 RUA needs those extensions to look inside the archives for 'tar_check' analysis."
-		);
-	} else {
-		env::set_var("PKGEXT", ".pkg.tar.xz");
-	};
-}
 
-pub fn extension() -> String {
-	std::env::var("PKGEXT").expect("Internal error: variable PKGEXT is unset")
+	let dirs = RuaDirs::new();
+	let mut pkgext = None;
+
+	let config = Command::new(&dirs.makepkg_config_loader)
+		.output()
+		.unwrap_or_else(|e| panic!("Reading makepkg config failed: {}", e))
+		.stdout;
+	let config = String::from_utf8(config).expect("makepkg config contains non-UTF-8 data");
+
+	// format: `VAR=VALUE\0`
+	let config_entries = config.split_terminator('\0').filter_map(|line| {
+		line.find('=')
+			.map(|sep_pos| (&line[..sep_pos], &line[sep_pos + 1..]))
+	});
+
+	// config entries won't appear here unless set
+	for (var, value) in config_entries {
+		debug!("makepkg option: {} = {}", var, value);
+		match var {
+			"PKGDEST" | "SRCDEST" | "SRCPKGDEST" | "LOGDEST" | "BUILDDIR" => {
+				panic!("Cannot work with {} set. Please run RUA without it", var)
```

Comment:
Now that we also read configuration files, should we downgrade this
to a warning, maybe?

Reply:
Yeah, I guess you're right.
It was very easy to start with the mindset "instantly bail if something might not meet user-s expectations". E.g. instead of working around some difficulties, only really work in the happy scenario. But here, the option to still continue with a warnings seems better suited than completely breaking, if you think about it.

tl&dr; agree, let's change that.